### DISCUSSION TITLE
Bluetooth bridge

### DISCUSSION THREAD

**Discussion by petrusek** [intelino-code/intelino-trainlib-async-py#2] [Ideas]
As suggested by @guoruicodes at #1 we could have a bluetooth bridge to the train to enable some AWS IoT functionality.

It is unclear whether it needs to be a mobile app, or it could be just a web-service published by a python script (as part of this library).

Let's explore what do we need to unlock this.

**Comment by zdila** [comment]
Maybe possible with Web Bluetooth API (Chrome on Android)?

**Reply by guoruicodes** [reply]
Connecting cloud -> python on laptop -> intelino python package -> ble > train could work as a dev setup / POC - I think if we check the AWS blogs they will have an IOT dev setup we can probably adapt. It could be lots of fun to add an alexa skill - "alexa start the train". I'm going to park this thought process for now (personally) as I still have a lot of learning to do about the app and the nice python library.

**Comment by guoruicodes** [comment]
Just dumping some more thoughts - the alexa skill would be nice as alexa can be used by children who can't yet fully use the app. Voice APIs are I think something that are going to become more an more prevalent and are a super way to interface with computers. Talking to computers will start once speech starts in a child. And what better computer to talk to than a (wooden) toy train set?

**Comment by petrusek** [comment]
Yes, voice commands are a great idea to use with the train.
There are speech libraries that can be used directly from Python (on a computer) without the need for a special bridge. One very nice example is here https://twitter.com/chronakis/status/1475426176135053320
